planet/clients/data.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2022 Planet Labs, PBC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+"""Functionality for interacting with the data api"""
+import logging
+import typing
+
+from .. import exceptions
+from ..constants import PLANET_BASE_URL
+from ..http import Session
+from ..models import Paged, Request, Response
+
+BASE_URL = f'{PLANET_BASE_URL}/data/v1/'
+SEARCHES_PATH = '/searches'
+STATS_PATH = '/stats'
+
+STATS_INTERVAL = ('hour', 'day', 'week', 'month', 'year')
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Items(Paged):
+    '''Asynchronous iterator over items from a paged response.'''
+    LINKS_KEY = '_links'
+    NEXT_KEY = '_next'
+    ITEMS_KEY = 'features'
+
+
+class DataClient():
+    """High-level asynchronous access to Planet's data API.
```

Comment:
@jreiberkyle wouldn't it be more correct to say "low level"? We're returning raw API JSON responses right now and that's about as low as we can get. High level, to me, implies Search, Stats, &c classes.

Reply:
yep, fully agree